SeaPublicWebsite/Controllers/CookieController.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using GovUkDesignSystem.GovUkDesignSystemComponents;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using SeaPublicWebsite.Helpers;
+using SeaPublicWebsite.Models.Cookies;
+using SeaPublicWebsite.Services;
+using SeaPublicWebsite.Services.Cookies;
+
+namespace SeaPublicWebsite.Controllers;
+
+public class CookieController: Controller
+{
+    private readonly CookieService cookieService;
+    private readonly GlobalConstants globalConstants;
+
+    public CookieController(CookieService cookieService, GlobalConstants globalConstants)
+    {
+        this.cookieService = cookieService;
+        this.globalConstants = globalConstants;
+    }
+
+    [HttpGet("/cookies")]
+    public IActionResult CookieSettings_Get(bool changesHaveBeenSaved = false)
+    {
+        cookieService.TryGetCookie<CookieSettings>(Request, cookieService.Configuration.CookieSettingsCookieName, out var cookie);
+        
+        var viewModel = new CookieSettingsViewModel
+        {
+            GoogleAnalytics = cookie?.GoogleAnalytics is true,
+            ChangesHaveBeenSaved = changesHaveBeenSaved
+        };
+        TempData["BannerState"] = BannerState.Hide;
+        return View("CookieSettings", viewModel);
+    }
+
+    [HttpPost("/cookies")]
+    [ValidateAntiForgeryToken]
+    public IActionResult CookieSettings_Post(CookieSettingsViewModel viewModel)
+    {
+        var cookieSettings = new CookieSettings
+        {
+            Version = cookieService.Configuration.CurrentCookieMessageVersion,
+            GoogleAnalytics = viewModel.GoogleAnalytics,
+        };
+        cookieService.SetCookie(Response, cookieService.Configuration.CookieSettingsCookieName, cookieSettings);
+        return CookieSettings_Get(changesHaveBeenSaved: true);
+    }
+
+    [HttpPost("/cookie-consent")]
+    [ValidateAntiForgeryToken]
+    public IActionResult CookieConsent(CookieConsent cookieConsent)
```

Comment:
Can you please check that all view model classes have `ViewModel` at the end of their names?

Reply:
Done! All I found was one file name typo.